server.go

```diff
@@ -296,13 +299,13 @@ func NewFromConfig(conf Config) (ret Server, err error) {
 		ret.traceBackend = &internalTraceBackend{}
 		ret.TraceClient, err = trace.NewBackendClient(ret.traceBackend, trace.Capacity(200))
 		if err != nil {
-			return
+			return ret, err
 		}
 
 		// configure Datadog as a Span sink
 		if conf.DatadogTraceAPIAddress != "" {
 
-			ddSink, err := NewDatadogSpanSink(&conf, ret.Statsd, ret.HTTPClient, ret.TagsAsMap)
+			ddSink, err := NewDatadogSpanSink(conf, ret.Statsd, ret.HTTPClient, ret.TagsAsMap)
```

Comment:
would like if this just passed a regular config struct value, not a pointer; nothing here is perf critical & it looks nicer.